openghg/util/_data_level.py

```diff
@@ -0,0 +1,58 @@
+from typing import Optional, overload
+import logging
+from openghg.types import MetadataFormatError
+
+__all__ = ["format_data_level"]
+
+logger = logging.getLogger("openghg.util")
+logger.setLevel(logging.INFO)  # Have to set level for logger as well as handler
+
+
+@overload
+def format_data_level(
+    data_level: str,
+) -> str: ...
+
+
+@overload
+def format_data_level(
+    data_level: None,
+) -> None: ...
+
+
+def format_data_level(data_level: Optional[str]) -> Optional[str]:
+    """
+    Check the input data_level to ensure this is in the expected format.
+
+    This should follow the convention of:
+        - "0": raw sensor output
+        - "1": automated quality assurance (QA) performed
+        - "2": final data set
+        - "3": elaborated data products using the data
+
+    Currently this will also allow 'decimal' inputs to indicate sub-levels (if required).
+    This can be used as an alternative to the data_sublevel key.
+
+    Args:
+        data_level: Specified data level for data
+    Returns:
+        str: Formatted
+    """
+
+    msg_expected = "Expect: '0', '1', '2', '3' (or decimal to indicate sub-level)"
+
+    if data_level is not None:
+        data_level = str(data_level)
```

Comment:
It was actually for a silly corner case which does come up where even though our type hinting asks for a string it doesn't enforce it so sometimes other types do get passed - this was to account for the integer case.
```suggestion
        data_level = str(data_level)  # in case a int/float is passed
```